src/lib/operators/aggregate_hash.hpp

```diff
@@ -80,8 +127,15 @@ using AggregateKeyEntry = uint64_t;
 // A dummy type used as AggregateKey if no GROUP BY columns are present
 struct EmptyAggregateKey {};
 
+// Used to store AggregateKeys if more than 2 GROUP BY columns are used
+constexpr auto AGGREGATE_KEY_SMALL_VECTOR_SIZE = 4;
```

Comment:
Added a comment.

Reply:
> Used to store AggregateKeys if more than 2 GROUP BY columns are used. The size is a trade-off between the memory consumption in the AggregateKeys vector (which becomes more expensive to read) and the cost of heap lookups. It could also be 5, which would be better for queries with 5 GROUP BY columns but worse for queries with 3 or 4.